bundles/org.eclipse.equinox.p2.publisher/src/org/eclipse/equinox/p2/publisher/actions/JREAction.java

```diff
@@ -70,8 +70,7 @@ public JREAction(String environment) {
 	@Override public IStatus perform(IPublisherInfo publisherInfo, IPublisherResult results, IProgressMonitor monitor) {
 		String problemMessage = NLS.bind(Messages.message_problemsWhilePublishingEE, jreLocation != null ? jreLocation : environment);
 		resultStatus = new MultiStatus(Activator.ID, 0, problemMessage, null);
-
-		initialize(publisherInfo);
+		this.info = publisherInfo;
```

Comment:
You're just bypassing the rather large initialization method.  Why?  That seems bad...

Reply:
I see, you are calling initialize elsewhere.